pkg/controller/gitopsconfig/controller.go

```diff
@@ -417,3 +438,169 @@ func isOwner(owner, owned metav1.Object) bool {
+		},
+		&jobOutput)
+
+	if err != nil {
+		log.Info("Error in GetJob for CheckResult:" + err.Error())
+		return jobState, err
+	}
+
+	jobState.State = getJobStatus(jobOutput.Status.Succeeded, jobOutput.Status.Failed, jobOutput.Status.Active)
+	if jobOutput.Status.Active > 0 && jobOutput.Status.Failed > 0 {
+		jobState.Message = "job failed, retrying..."
+	}
+	if jobOutput.Status.StartTime != nil {
+		jobState.StartTime = jobOutput.Status.StartTime.Format(time.RFC3339)
+	}
+	if jobOutput.Status.CompletionTime != nil {
+		jobState.CompletionTime = jobOutput.Status.CompletionTime.Format(time.RFC3339)
+	}
+
+	return jobState, nil
+}
+
+func (job *jobdata) watchjob(instance *gitopsv1alpha1.GitOpsConfig) {
+	log.Info("Watching the Job: " + job.Name)
+	var counter int
+	for {
+		time.Sleep(10 * time.Second)
+		jobState, err := job.checkResult()
+		if err != nil {
+			continue
+		}
+		instance.Status = jobState
+		if jobState.State != "INPROGRESS" {
+			instance.Status.CompletionTime = jobState.CompletionTime
+			instance.Status.State = jobState.State
+			err := job.updateStatus(instance)
+			if err != nil {
+				log.Info("Error in Updating the job when jobState.Active == 0:" + err.Error())
+			}
+			return
+		}
+
+		if counter == 0 {
+			instance.Status.State = jobState.State
+			instance.Status.CompletionTime = ""
+			log.Info("Intial Update")
+			err := job.updateStatus(instance)
+			if err != nil {
+				log.Info("Error in Intial Update of job Status:" + err.Error())
+			}
+		}
+
+		counter++
+	}
+}
+
+func (job *jobdata) updateStatus(instance *gitopsv1alpha1.GitOpsConfig) error {
+
+	err := job.Reconcile.client.Status().Update(context.TODO(), instance)
+	if err != nil {
+		log.Info("Job name:" + job.Name)
+		log.Info("Error in UpdateStatus.Update():" + err.Error())
+		return err
+	}
+	log.Info("Status updated for job : " + job.Name)
+	return nil
+}
+
+func watchJobForStatus(r *Reconciler, jobName string, namespace string, instance gitopsv1alpha1.GitOpsConfig) {
+	var job = jobdata{
+		Reconcile: r,
+		Name:      jobName,
+		Namespace: namespace,
+	}
+	job.watchjob(&instance)
+}
+
+func scheduleStatusForCronJobs(job jobdata, schedule string, instance *gitopsv1alpha1.GitOpsConfig) {
+	var result batchv1beta1.CronJob
+	err := job.Reconcile.client.Get(context.TODO(),
+		types.NamespacedName{Name: job.Name, Namespace: job.Namespace},
+		&result)
+
+	if err != nil {
+		log.Info("Error in GetCronJob:" + err.Error())
+		return
+	}
+
+	instance.Status = gitopsv1alpha1.GitOpsConfigStatus{}
+	if len(result.Status.Active) < 1 {
+		log.Info("There is no active job")
+		if result.Status.LastScheduleTime != nil {
+			instance.Status.LastScheduleTime = result.Status.LastScheduleTime.Format(time.RFC3339)
+		}
+		instance.Status.Message = "There is no active job"
+		err = job.Reconcile.client.Status().Update(context.TODO(), instance)
+		if err != nil {
+			log.Info("Error in Updating the status for cron job:" + err.Error())
+		}
+	}
+	c := cron.New()
```

Comment:
I believe we should add a line above with:

    // TODO: convert from cron to an approach based on Watch, see "Creating watches" in: https://blog.openshift.com/kubernetes-operators-best-practices/